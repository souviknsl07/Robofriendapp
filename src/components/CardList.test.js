import { render } from "enzyme";
import React from "react";
import CardList from "./CardList";

it("expect to render Card component", () => {
  const mockRobots = [
    {
      id: 1,
      name: "John Snow",
      username: "JohnJohn",
      email: "john@gmail.com",
    },
  ];
  expect(render(<CardList robot={mockRobots} />)).toMatchSnapshot();
});
