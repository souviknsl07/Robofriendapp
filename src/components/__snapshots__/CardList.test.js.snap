// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
initialize {
  "0": Node {
    "attribs": Object {},
    "children": Array [
      Node {
        "attribs": Object {
          "class": "tc bg-light-green dib br3 pa2 ma2 grow bw2 shadow-5",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "alt": "robots",
              "src": "https://robohash.org/1?size=200x200",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {},
              "children": Array [
                Node {
                  "attribs": Object {},
                  "children": Array [
                    Node {
                      "data": "John Snow",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "h2",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Node {
                    "attribs": Object {},
                    "children": Array [
                      Node {
                        "data": "john@gmail.com",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "p",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {},
                    "x-attribsPrefix": Object {},
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
                Node {
                  "attribs": Object {},
                  "children": Array [
                    Node {
                      "data": "john@gmail.com",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "p",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Node {
                    "attribs": Object {},
                    "children": Array [
                      Node {
                        "data": "John Snow",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "h2",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {},
                    "x-attribsPrefix": Object {},
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "src": undefined,
            },
          },
          Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "attribs": Object {},
                "children": Array [
                  Node {
                    "data": "John Snow",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "h2",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Node {
                  "attribs": Object {},
                  "children": Array [
                    Node {
                      "data": "john@gmail.com",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "p",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              Node {
                "attribs": Object {},
                "children": Array [
                  Node {
                    "data": "john@gmail.com",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {},
                  "children": Array [
                    Node {
                      "data": "John Snow",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "h2",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "alt": "robots",
                "src": "https://robohash.org/1?size=200x200",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "alt": undefined,
                "src": undefined,
              },
              "x-attribsPrefix": Object {
                "alt": undefined,
                "src": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
